projects/components/src/table/renderers/table-cell-renderer.component.ts

```diff
@@ -12,31 +11,59 @@ import {
 
 @Directive() // Angular 9 Requires superclasses to be annotated as well in order to resolve injectables for constructor
 // tslint:disable-next-line:directive-class-suffix
-export abstract class TableCellRendererComponent<TRaw, TParsed = TRaw> implements OnInit {
+export abstract class TableCellRendererComponent<TCellData, TValue = TCellData> implements OnInit {
   public static readonly type: StandardTableCellRendererType;
   public static readonly alignment: TableCellAlignmentType;
 
-  private _value!: TParsed;
+  private _value!: TValue;
+  private _units!: string;
+  private _tooltip!: string;
   public readonly clickable: boolean = false;
   public readonly isFirstColumn: boolean = false;
 
-  protected abstract parseValue(cellData?: TRaw, rowData?: Dictionary<unknown>): TParsed;
-
   public constructor(
     @Inject(TABLE_CELL_RENDERER_COLUMN_CONFIG) private readonly columnConfig: TableColumnConfig,
     @Inject(TABLE_CELL_RENDERER_COLUMN_INDEX) private readonly index: number,
-    @Optional() @Inject(TABLE_CELL_RENDERER_CELL_DATA) private readonly cellData: TRaw | null,
-    @Optional() @Inject(TABLE_CELL_RENDERER_ROW_DATA) private readonly rowData: Dictionary<unknown> | null
+    @Inject(TABLE_CELL_RENDERER_ROW_DATA) private readonly rowData: TableRow,
+    @Inject(TABLE_CELL_RENDERER_CELL_DATA) private readonly cellData: TCellData
   ) {
     this.clickable = this.columnConfig.onClick !== undefined;
     this.isFirstColumn = this.index === 0;
   }
 
   public ngOnInit(): void {
-    this._value = this.parseValue(this.cellData ?? undefined, this.rowData ?? undefined);
+    this.initialize();
+  }
+
+  public initialize(): void {
+    this._value = this.parseValue(this.cellData, this.rowData);
+    this._units = this.parseUnits(this.cellData, this.rowData);
+    this._tooltip = this.parseTooltip(this.cellData, this.rowData);
+  }
+
+  protected abstract parseValue(cellData: TCellData, rowData?: TableRow): TValue;
+
+  protected parseUnits(_cellData: TCellData, _rowData?: TableRow): string {
+    return '';
```

Comment:
is this intentionally left empty?

Reply:
Yep. Many attributes don't have units, so this is sort of a no-op to build the default tooltip.